lib/backup_restore/restorer.rb

```diff
@@ -195,8 +195,15 @@ def sidekiq_has_running_jobs?
     end
 
     def copy_archive_to_tmp_directory
-      log "Copying archive to tmp directory..."
-      Discourse::Utils.execute_command('cp', @source_filename, @archive_filename, failure_message: "Failed to copy archive to tmp directory.")
+      if @store.remote?
+        log "Downloading archive to tmp directory..."
+        failure_message = "Failed to download archive to tmp directory."
+      else
+        log "Copying archive to tmp directory..."
+        failure_message = "Failed to copy archive to tmp directory."
+      end
```

Comment:
```
failure_message =
  if @store.remote?
  else
  end
```
We can do something like this too.

Reply:
It might be me, but I find it a lot easier to read with the explicit assignments. I'd like to keep it that way.

Reply:
:+1: